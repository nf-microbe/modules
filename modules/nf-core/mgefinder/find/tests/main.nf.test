nextflow_process {

    name "Test Process MGEFINDER_FIND"
    script "../main.nf"
    process "MGEFINDER_FIND"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "mgefinder"
    tag "mgefinder/formatbam"
    tag "mgefinder/find"

    test("fasta.gz - bam") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    [
                        file("/workspaces/modules/.nf-test/tests/b9e55730919a2a80ba7dab72595ea208/work/74/1f51f7f03ef3fb03744d3d337c38a6/test.mgefinder.bam", checkIfExists: true),
                        file("/workspaces/modules/.nf-test/tests/b9e55730919a2a80ba7dab72595ea208/work/74/1f51f7f03ef3fb03744d3d337c38a6/test.mgefinder.bam.bai", checkIfExists: true)
                    ]
                ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("bam - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    []
                ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
