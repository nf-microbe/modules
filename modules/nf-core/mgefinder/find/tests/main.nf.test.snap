{
    "bam - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.mgefinder.find:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,9da89ee786595ead7f36484381f71f90"
                ],
                "find": [
                    [
                        {
                            "id": "test"
                        },
                        "test.mgefinder.find:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,9da89ee786595ead7f36484381f71f90"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-07T22:04:07.07098309"
    },
    "fasta.gz - bam": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.mgefinder.find:md5,11ff91453ea6e51a1a2641ff4dd1aeda"
                    ]
                ],
                "1": [
                    "versions.yml:md5,9da89ee786595ead7f36484381f71f90"
                ],
                "find": [
                    [
                        {
                            "id": "test"
                        },
                        "test.mgefinder.find:md5,11ff91453ea6e51a1a2641ff4dd1aeda"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,9da89ee786595ead7f36484381f71f90"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-07T22:46:12.933252031"
    }
}